PyRDF/backend/Local.py

```diff
@@ -63,6 +63,9 @@ def execute(self, generator):
         values[0].GetValue() # Trigger event-loop
 
         for i in range(len(values)):
-            # Set the values of action nodes to
-            # 'TResultPtr's obtained
-            nodes[i].value = values[i]
\ No newline at end of file
+            # The type of the value of the action node
+            value_type = type(values[i].GetValue())
+            # The `value_type` is required here because,
+            # after a call to `GetValue`, the values die
+            # along with the RResultPtrs
+            nodes[i].value = value_type(values[i].GetValue())
```

Comment:
Why do you need to copy construct it? Can't you just do:
`nodes[i].value = values[i].GetValue())` ?

Reply:
@etejedor , the same reason as spark backend. It seems to throw the same error workout that

Reply:
Is it because the `RResultPtr`s are going out of scope and thus garbarge collected, together with their values?
In Spark we need to copy construct because we cannot serialize `RResultPtrs`. In the local backend, we do not need to serialize them, so if we want to avoid the copy we just need to keep a reference to them somewhere.

Reply:
@etejedor yes. `RResultPtr`s are going out of scope without the copy construction I believe. This is the error that pops up without copy construction : `AttributeError: 'PyROOT_NoneType' object has no attribute 'Draw' `. That is for calling a `Draw` on a Histo object

Reply:
@etejedor , I'm guessing that this is because I'm constructing the rdf object and triggering the event-loop within the `execute` method of Local. So, may be `RResultPtr`s get out of scope after execution when we return back to the `Proxy` object from where `execute` was called

Reply:
@etejedor and @bluehood , any fix for this ?

Reply:
To avoid the copy in the local backend case, I suggest we keep storing the `RResultPtr`s. Then, in the logic of `GetValue` in `Proxy`, we can invoke `GetValue` on the stored object if it is an `RResultPtr`, otherwise we just returned the stored value.
This can also be implemented with a `try` `except`, returning `GetValue` in the try and returning the stored value in the `except`.

Reply:
@etejedor , but then it wouldn't be consistent across all backends then. Is that okay ?

Reply:
Another option is to store both the RResultPtr and the actual values. In the case of the Spark backend, it would only set the actual values. In the case of the local backend, it would set both. Keeping the RResultPtrs would just prevent the values from disappearing.

Reply:
@etejedor , I don't understand this. Could you elaborate ?

Reply:
@etejedor , resolved this now 👍 
Sorry for the delay